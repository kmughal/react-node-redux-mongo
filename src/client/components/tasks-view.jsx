import React from "react";
import PropTypes from "prop-types";
let count = 0;

export const TasksViewComponent = ({ tasks }) => (
	<section>
		<button
			className="btn btn-success"
			type="button"
			data-toggle="collapse"
			data-target="#tasks-list"
			aria-expanded="false"
			aria-controls="collapseExample"
		>
			Tasks [{tasks.length}]
		</button>
	
		<div>
			<br/>
			{tasks.length ? (
				<div className="mt-10 pt-20">
					{tasks.map(task => (
						<div className="card p-2 mt-10" key={count++}>
							<h5>A task added by you.</h5>
							<div className="card-body">
								<p className="card-text">{task.task}</p>
							</div>
							<button className="btn btn-primary">Edit</button>
						</div>
					))}
				</div>
			) : (
				<p className="alert alert-info" role="alert">
					Currently you dont have any task.
				</p>
			)}
		</div>
	</section>
);

TasksViewComponent.propTypes = {
	tasks: PropTypes.array
};
