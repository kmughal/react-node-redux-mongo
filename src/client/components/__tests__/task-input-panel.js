import {
  mapStateToProps
} from "../task-input-panel";
import React from "react" ;
import renderer from "react-test-renderer";
import {TasksViewComponent} from "../tasks-view";

describe("The task input panel component.", () => {
  describe("mapStateToProps", () => {
    it("Should map states to props correctly", () => {
      const tasks = {
        tasks: [{
          task: "TODO"
        }]
      };
      const appState = {
        records: {
          tasks: [{
            task: "TODO"
          }]
        },
        sessions: {}
      };
      const props = {};
      const componentState = mapStateToProps(appState, props);
      expect(componentState).toEqual(tasks);
    })
  });

  describe("TaskInputPanelComponent" , ()=> {
    describe("TasksViewComponent" , ()=> {
      const tasks =  [{
        task: "TODO!"
      }];
      const tree = renderer.create(<TasksViewComponent tasks={tasks}/>);
      expect(tree.toJSON()).toMatchSnapshot();
    })
  })
});