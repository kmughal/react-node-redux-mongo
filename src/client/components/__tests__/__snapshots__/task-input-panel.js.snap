// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<section>
  <button
    aria-controls="collapseExample"
    aria-expanded="false"
    className="btn btn-success"
    data-target="#tasks-list"
    data-toggle="collapse"
    type="button"
  >
    Tasks [
    1
    ]
  </button>
  <div>
    <br />
    <div
      className="mt-10 pt-20"
    >
      <div
        className="card p-2 mt-10"
      >
        <h5>
          A task added by you.
        </h5>
        <div
          className="card-body"
        >
          <p
            className="card-text"
          >
            TODO!
          </p>
        </div>
        <button
          className="btn btn-primary"
        >
          Edit
        </button>
      </div>
    </div>
  </div>
</section>
`;
